FIX: Adding use_scalebars as a parameter.

adam2392 opened the pull request:
Adds "use_scalebars" as a boolean parameter into plot() and plot_raw().

#### Reference issue
Example: Fixes #7068 

#### What does this implement/fix?
Added a boolean parameter to allow programmatic on/off of the unit scale bars.

adam2392 commented:
Example of it being done.

[plot_rawdata_scalebars.pdf](https://github.com/mne-tools/mne-python/files/3870235/plot_rawdata_scalebars.pdf)

drammock commented on mne/viz/raw.py:
```diff
@@ -235,6 +235,10 @@ def plot_raw(raw, events=None, duration=10.0, start=0.0, n_channels=20,
         .. versionadded:: 0.16.0
+    use_scalebars : bool | True
+        Whether, or not to show the scale bars. Defaults to True.
+        .. vertionadded:: 0.19.0
```
vertion -> version

drammock commented on mne/io/base.py:
```diff
@@ -1558,13 +1558,13 @@ def plot(self, events=None, duration=10.0, start=0.0, n_channels=20,
              show_first_samp=False, proj=True, group_by='type',
              butterfly=False, decim='auto', noise_cov=None, event_id=None,
-             show_scrollbars=True, verbose=None):
+             show_scrollbars=True, use_scalebars=True, verbose=None):
```
I think "show_scalebars" is more consistent with how we name other parameters, and is more specific too ("use" is a rather generic verb)

adam2392 replied:
Wanted to keep the wording similar. So I use the "use_scalebars" because in _plot_raw_traces, the parameter is named "use_scalebars". Should I still make this change?

drammock commented on mne/viz/raw.py:
```diff
@@ -235,6 +235,10 @@ def plot_raw(raw, events=None, duration=10.0, start=0.0, n_channels=20,
         theh event numbers).
 
         .. versionadded:: 0.16.0
+    use_scalebars : bool | True
```
This should just be "bool" not "bool | True".

drammock commented on mne/viz/raw.py:
```diff
@@ -235,6 +235,10 @@ def plot_raw(raw, events=None, duration=10.0, start=0.0, n_channels=20,
 
         .. versionadded:: 0.16.0
+    use_scalebars : bool | True
+        Whether, or not to show the scale bars. Defaults to True.
```
The comma after "whether" is unnecessary

GuillaumeFavelier commented:
>  I think there might even be a "make style" directive that runs them all

`make pep` should do the trick.

More details in `mne-python` [Makefile](https://github.com/mne-tools/mne-python/blob/master/Makefile#L131-L132).

adam2392 commented:
I'm getting an error when running "make pep". Not sure what is going on, since I fixed the above issues. I don't see what the issue is, but maybe I'm missing something obvious.

`    @requires_numpydoc
    def test_docstring_parameters():
        """Test module docstring formatting."""
        from numpydoc import docscrape
    
        # skip modules that require mayavi if mayavi is not installed
        public_modules_ = public_modules[:]
        try:
            import mayavi  # noqa: F401 analysis:ignore
            public_modules_.append('mne.gui')
        except ImportError:
            pass
    
        incorrect = []
        for name in public_modules_:
            # Assert that by default we import all public names with `import mne`
            if name not in ('mne', 'mne.gui'):
                extra = name.split('.')[1]
                assert hasattr(mne, extra)
            with pytest.warns(None):  # traits warnings
                module = __import__(name, globals())
            for submod in name.split('.')[1:]:
                module = getattr(module, submod)
            classes = inspect.getmembers(module, inspect.isclass)
            for cname, cls in classes:
                if cname.startswith('_') and cname not in _doc_special_members:
                    continue
                with pytest.warns(None) as w:
                    cdoc = docscrape.ClassDoc(cls)
                for ww in w:
                    if 'Using or importing the ABCs' not in str(ww.message):
                        raise RuntimeError('Error for __init__ of %s in %s:\n%s'
                                           % (cls, name, ww))
                if hasattr(cls, '__init__'):
                    incorrect += check_parameters_match(cls.__init__, cdoc, cls)
                for method_name in cdoc.methods:
                    method = getattr(cls, method_name)
                    incorrect += check_parameters_match(method, cls=cls)
                if hasattr(cls, '__call__'):
                    incorrect += check_parameters_match(cls.__call__, cls=cls)
            functions = inspect.getmembers(module, inspect.isfunction)
            for fname, func in functions:
                if fname.startswith('_'):
                    continue
                incorrect += check_parameters_match(func)
        msg = '\n' + '\n'.join(sorted(list(set(incorrect))))
        if len(incorrect) > 0:
>           raise AssertionError(msg)
E           AssertionError: 
E           mne.io.base.BaseRaw.plot show_scrollbars != use_scalebars
E           mne.io.base.BaseRaw.plot use_scalebars != show_scrollbars
E           mne.io.base.Raw.plot show_scrollbars != use_scalebars
E           mne.io.base.Raw.plot use_scalebars != show_scrollbars
E           mne.io.base.RawArray.plot show_scrollbars != use_scalebars
E           mne.io.base.RawArray.plot use_scalebars != show_scrollbars
E           mne.viz.raw.plot_raw show_scrollbars != use_scalebars
E           mne.viz.raw.plot_raw use_scalebars != show_scrollbars`

drammock commented:
This is from the Azure output:

```
E           mne.viz.raw.plot_raw : GL03 : Double line break found; please use only one blank line to separate sections or paragraphs, and do not leave blank lines at the end of docstrings
E           mne.viz.raw.plot_raw : PR03 : Wrong parameters order. Actual: ('raw', 'events', 'duration', 'start', 'n_channels', 'bgcolor', 'color', 'bad_color', 'event_color', 'scalings', 'remove_dc', 'order', 'show_options', 'title', 'show', 'block', 'highpass', 'lowpass', 'filtorder', 'clipping', 'show_first_samp', 'proj', 'group_by', 'butterfly', 'decim', 'noise_cov', 'event_id', 'show_scrollbars', 'use_scalebars', 'verbose'). Documented: ('raw', 'events', 'duration', 'start', 'n_channels', 'bgcolor', 'color', 'bad_color', 'event_color', 'scalings', 'remove_dc', 'order', 'show_options', 'title', 'show', 'block', 'highpass', 'lowpass', 'filtorder', 'clipping', 'show_first_samp', 'proj', 'group_by', 'butterfly', 'decim', 'noise_cov', 'event_id', 'use_scalebars', 'show_scrollbars', 'verbose')
```

I didn't realize that the internal parameter was called `use_scalebar`.  I still think `show_scalebar` is a better name for the public API, so I'd be inclined to change the internal variable name to match the (new) public parameter name.

cbrnr commented:
👍 for renaming to `show_scalebar`.

drammock commented:
@adam2392 you're still failing the CI tests; FYI generally we wait to make sure all tests are passing before asking for review.

The first error I looked at on Azure was definitely related to this PR.  Can you try to figure out what is failing, why it's failing, and fix it?  If you try and fail, that's OK, just let us know that someone else needs to take over.

adam2392 commented:
Ah yes so sorry about that! I was running the tests locally and I think I didn't run the full suite. Will wait for the full CI to finish before doing so again.

drammock commented:
cool.  Note that you can run (from the root of the MNE-Python folder) `pytest mne/viz/tests/test_epochs.py` (for example) if you want to run only the tests in a single file.  If your PR is a small one, this is *usually* sufficient to catch all the likely errors.  I recommend this, actually, since the full test suite takes a very long time to run locally. See https://mne.tools/dev/install/contributing.html#running-the-test-suite

Please ping me again when all tests are passing and I'll review at that time.

adam2392 commented:
Thanks for the tips @drammock ! I am hoping to make more contributions to MNE throughout the future, as it's been such a huge help for my research.

Looks like all the tests have passed. Lmk if there is anything else you would like me to change.

drammock commented on mne/viz/raw.py:
```diff
@@ -232,10 +232,14 @@ def plot_raw(raw, events=None, duration=10.0, start=0.0, n_channels=20,
+    show_scalebars : bool
+        Whether or not to show the scale bars. Defaults to True.
+
+        .. versionadded:: 0.19.0w
```
```suggestion
        .. versionadded:: 0.19.0
```

drammock commented on mne/viz/ica.py:
```diff
@@ -903,7 +903,8 @@ def _plot_sources_raw(ica, raw, picks, exclude, start, stop, show, title,
                   first_time=first_time, data_picks=[], decim=1,
                   noise_cov=None, whitened_ch_names=(), clipping=None,
-                  use_scalebars=False, show_scrollbars=show_scrollbars)
+                  use_scalebars=False, show_scrollbars=show_scrollbars,
```
```suggestion
                  show_scrollbars=show_scrollbars,
```

drammock replied:
looks like you forgot to remove `use_scalebars` here after the name change.

adam2392 commented:
Cool, thanks for that quick catch! Yep, this looks good to me, and the plots look fine upon using it.

larsoner commented:
Thanks @adam2392
